packages/client/src/helpers.ts

```diff
@@ -123,6 +125,27 @@ export function rethrowKnownErrorTypes(err: GrpcServiceError): void {
         const { namespace } = temporal.api.errordetails.v1.NamespaceNotFoundFailure.decode(entry.value);
         throw new NamespaceNotFoundError(namespace);
       }
+      case 'temporal.api.errordetails.v1.MultiOperationExecutionFailure': {
+        // MultiOperationExecutionFailure contains error statuses for multiple
+        // operations. A MultiOperationExecutionAborted error status means that
+        // the corresponding operation was aborted due to an error in one of the
+        // other operations. We rethrow the first operation error that is not
+        // MultiOperationExecutionAborted.
+        const { statuses } = temporal.api.errordetails.v1.MultiOperationExecutionFailure.decode(entry.value);
+        for (const status of statuses) {
+          const detail = status.details?.[0];
+          const statusType = detail?.type_url?.replace(/^type.googleapis.com\//, '') as FailureName | undefined;
+          if (statusType === 'temporal.api.failure.v1.MultiOperationExecutionAborted') {
+            continue;
+          }
+          throw new CustomGrpcServiceError(
```

Comment:
What kind of details do we have here? Could we map these statuses to error classes that we already have elsewhere?

Reply:
See above https://github.com/temporalio/sdk-typescript/pull/1585#discussion_r1890975131